numpy/f2py/__init__.py

```diff
@@ -59,10 +62,32 @@ def compile(source,
         f.write(source)
         f.flush()
 
-        args = ' -c -m {} {} {}'.format(modulename, f.name, extra_args)
-        c = '{} -c "import numpy.f2py as f2py2e;f2py2e.main()" {}'
-        c = c.format(sys.executable, args)
-        status, output = exec_command(c)
+        args = ['-c', '-m', modulename, f.name]
+
+        if os.name == 'posix':
+            posix = True
+        else:
+            posix = False
+
+        extra_args = shlex.split(extra_args, posix=posix)
```

Comment:
Might want to guard this with `if isinstance(extra_args, np.compat.basestring)`, to allow lists to be passed in future (and update the docs accordingly)